### PR TITLE
EIP-2628 - Header in StatusMessage

### PR DESCRIPTION
**Pull request by tkstanczak**
Adds a proposal for the status message change.

When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-2628.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+eip: eip-2628
+title: Header in StatusMessage
+author: Tomasz K. Stanczak (@tkstanczak)
+discussions-to: TBD
+status: Draft
+type: Standards Track
+category: Core
```
Comment (axic) [inline EIPS/eip-2628.md:8]: I believe this is `Networking`.

File: EIPS/eip-2628.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+eip: eip-2628
+title: Header in StatusMessage
```
Comment (axic) [inline EIPS/eip-2628.md:3]: I think this is going to be something like `eth/66: Header in StatusMessage` and the description then proposes a change to the existing protocol (eth/65?).

See https://eips.ethereum.org/EIPS/eip-2464 as an example.

@fjl @karalabe can you guys give some directions here?

File: EIPS/eip-2628.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Motivation
+
+Currently after a handshake with a new node, if the node has a better total difficulty block, then we need to ask for the block header to be able to verify the total difficulty claim.
+With this change this will be no longer required which will make the network slightly more resilient to sybil attacks.
+
+## Specification
+
+Remove bestHash, genesisHash, number fields from the StatusMessage. Add Rlp_noBloom(BlockHeader) instead.
+When calculating a block header hash use Keccak(Bloom) instead of Bloom before serializing the header to RLP (this is to decrease the Header size from ~512 bytes to ~288 bytes).
+Furthermore with this change we can decrease the storage size of headers from genesis by ~3GB for nodes that do not require historical bodies and receipts if we also change the BlockHeaders message to only optionally require blooms.
```
Comment (karalabe) [inline EIPS/eip-2628.md:29]: The `eth` protocol is meant for full nodes. If a node does not retain historical headers and bodies, they are actively harming network health and should not be on the `eth` protocol. I don't see a reason why we'd want to aid bad behavior.

File: EIPS/eip-2628.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+This document proposes to include the whole BlockHeader in the status message and change the way the header hash is calculated to decrease the required size of the StatusMessage in such scenario.
+
+## Abstract
+
+Remove number, bestHash, and genesisHash and include the full block header in the Eth62 StatusMessage. When calculating a header hash use Keccak(Bloom) instead of Bloom when serializing header.
+
+## Motivation
+
+Currently after a handshake with a new node, if the node has a better total difficulty block, then we need to ask for the block header to be able to verify the total difficulty claim.
+With this change this will be no longer required which will make the network slightly more resilient to sybil attacks.
```
Comment (karalabe) [inline EIPS/eip-2628.md:23]: I don't really see why sending a header is more secure then sending number/hash. You can still create a fake header for free, mine it with insignificant PoW and say that it's header 20 million with TD infinite. You don't gain any security by sending more data, or at least I don't see how.

File: EIPS/eip-2628.md (hunk 2 above)
Comment (karalabe) [reply EIPS/eip-2628.md:3]: There's already an eth/66 from @cburgdorf

Comment (cburgdorf) [reply EIPS/eip-2628.md:3]: Just linking to the [`eth/ 66` EIP](https://github.com/ethereum/EIPs/pull/2481/files)

**Comment by tkstanczak** [comment]
It is only 'slightly' more resilient to sybil attacks because the time to
learn about good blocks is shortened on non-malicious nodes. It is not
about preventing forged blocks.

File: EIPS/eip-2628.md (hunk 2 above)
Comment (axic) [reply EIPS/eip-2628.md:3]: https://eips.ethereum.org/EIPS/eip-2481 aka "eth/66" was merged, so I guess this is "eth/67" then?

File: EIPS/eip-2628.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+---
+eip: eip-2628
```
Comment (axic) [inline EIPS/eip-2628.md:2]:
```suggestion
eip: 2628
```

File: EIPS/eip-2628.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+## Rationale
+
+It will be easier to quickly verify best branch claims from other nodes.
+It may open ways to decrease the headers storage size (minor priority but it can improve stateless clients requirements)
+
+## Backwards Compatibility
+
+Not backward compatible. May require changes in tooling that is calculating / verifying header hashes.
+
```
Comment (axic) [inline EIPS/eip-2628.md:39]: Missing Security Considerations section.

**Pull request closed without merging**